shopinvader_customer_validate/migrations/14.0.1.2.3/pre-migration.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from openupgradelib import openupgrade
+
+# pylint: disable=odoo-addons-relative-import
+from odoo.addons.shopinvader_customer_validate.hooks import (
+    NEW_MODULE,
+    OLD_MODULE,
+    RECORD_XMLIDS,
+    update_record_xmlids,
+)
+
+
+def check_installed(cr):
+    query = """
+        SELECT 1
+        FROM ir_model_data
+        WHERE module = %(old_module)s
+        AND name in %(names)s;
+    """
+    cr.execute(query, {"old_module": OLD_MODULE, "names": tuple(RECORD_XMLIDS)})
+    return cr.fetchone()
+
+
+def delete_new_records(cr):
+    xmlids = [f"{NEW_MODULE}.{xmlid}" for xmlid in RECORD_XMLIDS]
+    openupgrade.delete_records_safely_by_xml_id(xmlids)
```

Comment:
I'm getting an error due to this being missing here.

```suggestion
env = api.Environment(cr, SUPERUSER_ID, {})
openupgrade.delete_records_safely_by_xml_id(env, xmlids)
```





But even after fixing that, the script is not working on my db 😭 


The thing is we can't just make the assumption to delete the "new records", because some of them may actually be used.. since we didn't have the migration script at the right time.

**Silent error during `delete_records_safely_by_xml_id` (because of the "safely" part)**
```
Error deleting XML-ID shopinvader_customer_validate.email_address_validated_notification: IntegrityError('update or delete on table "mail_template" violates foreign key constraint "shopinvader_notification_template_id_fkey" on table "shopinvader_notification"\nDETAIL:  Key (id)=(99) is still referenced from table "shopinvader_notification".\n') 
```

**Error during `update_record_xmlids`**
```
psycopg2.IntegrityError: duplicate key value violates unique constraint "ir_model_data_module_name_uniq_index"
DETAIL:  Key (module, name)=(shopinvader_customer_validate, email_new_customer_welcome_not_validated_notification) already exists.
```